cmd/weaver-kube/deploy.go

```diff
@@ -147,3 +152,42 @@ func deploy(ctx context.Context, args []string) error {
 	// Generate the kube deployment information.
 	return impl.GenerateKubeDeployment(image, dep, config)
 }
+
+// checkVersionCompatibility checks that the tool binary is compatible with
+// the application binary being deployed.
+func checkVersionCompatibility(appBinary string) error {
+	versions, err := bin.ReadVersions(appBinary)
+	if err != nil {
+		return fmt.Errorf("read versions: %w", err)
+	}
+	selfVersion, _, err := impl.ToolVersion()
+	if err != nil {
+		return fmt.Errorf("read weaver-kube version: %w", err)
+	}
+	if versions.DeployerVersion != version.DeployerVersion {
+		// Try to relativize the binary, defaulting to the absolute path if
+		// there are any errors..
+		binary := appBinary
+		if cwd, err := os.Getwd(); err == nil {
```

Comment:
would this become easier to parse if we put it in a function?

relativizeBinaryFn := func() string {
   cwd, err := os.Getwd()
   if err != nil {
     return appBinary
   }
   rel, err := filepath.Rel(cwd, appBinary)
   if err == nil {
     return rel
   }
   return appBinary
}

binary := relativizeBinaryFn()

Reply:
Thanks, done!